File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var r1 = -1
        var r2 = -1
        val map = mutableMapOf<Int, Int>()

        for (i in nums.indices) {
            val num1 = nums[i]
            val num2 = target - num1
            val j = map[num2]

            if (j != null) {
                r1 = i
                r2 = j
                break
            }

            map[num1] = i
        }


        return intArrayOf(r1, r2)
    }
}

File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut result = Vec::new();
    let mut map: HashMap<i32, usize> = HashMap::new();

    for (i, num) in nums.iter().enumerate() {
        let find = target - num;
        let try_finding = map.get(&find);

        match try_finding {
            Some(j) => {
                result.push(i as i32);
                result.push(*j as i32);
                break;
            }
            None => {
                map.insert(*num, i);
            }
        }
    }

    return result;
}


}